features/mbedtls/targets/TARGET_Silicon_Labs/crypto_sha.c

```diff
@@ -0,0 +1,461 @@
+    0x1F83D9ABUL,
+    0x5BE0CD19UL
+};
+
+static const uint32_t init_state_sha224[8] =
+{
+    0xC1059ED8UL,
+    0x367CD507UL,
+    0x3070DD17UL,
+    0xF70E5939UL,
+    0xFFC00B31UL,
+    0x68581511UL,
+    0x64F98FA7UL,
+    0xBEFA4FA4UL
+};
+#endif /* defined(MBEDTLS_SHA256_C) */
+
+static const unsigned char sha_padding[64] =
+{
+ 0x80, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+    0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0
+};
+
+typedef enum {
+#if defined(MBEDTLS_SHA1_C)
+    CRYPTO_SHA1,
+#endif /* defined(MBEDTLS_SHA1_C) */
+#if defined(MBEDTLS_SHA256_C)
+    CRYPTO_SHA2
+#endif /* defined(MBEDTLS_SHA256_C) */
+} crypto_sha_mode_t;
+
+/*
+ * 32-bit integer manipulation macros (big endian)
+ */
+#ifndef PUT_UINT32_BE
+#define PUT_UINT32_BE(n,b,i)                            \
+do {                                                    \
+    (b)[(i)    ] = (unsigned char) ( (n) >> 24 );       \
+    (b)[(i) + 1] = (unsigned char) ( (n) >> 16 );       \
+    (b)[(i) + 2] = (unsigned char) ( (n) >>  8 );       \
+    (b)[(i) + 3] = (unsigned char) ( (n)       );       \
+} while( 0 )
+#endif
+
+/**
+ * @brief unified SHA acceleration function.
+ *
+ * @param[inout] state  State context for SHA hashing
+ * @param[in] data      Input block(s). Must contain N blocks
+ *                      of SHA1/SHA256 block size (64 bytes)
+ * @param blocks        Amount of blocks pointed to by data
+ * @param mode          SHA operation mode
+ */
+static void crypto_sha_update_state( uint32_t state[8],
+                                     const unsigned char *data,
+                                     size_t blocks,
+                                     crypto_sha_mode_t mode )
+{
+    CORE_DECLARE_IRQ_STATE;
+    CRYPTO_TypeDef *crypto = crypto_management_acquire();
+
+    switch ( mode ) {
+#if defined(MBEDTLS_SHA1_C)
+        case CRYPTO_SHA1:
+            crypto->CTRL = CRYPTO_CTRL_SHA_SHA1;
+            break;
+#endif /* defined(MBEDTLS_SHA1_C) */
+#if defined(MBEDTLS_SHA256_C)
+        case CRYPTO_SHA2:
+            crypto->CTRL = CRYPTO_CTRL_SHA_SHA2;
+            break;
+#endif /* defined(MBEDTLS_SHA256_C) */
+    }
+
+    crypto->WAC      = 0;
+    crypto->IEN      = 0;
+
+    /* Set result width of MADD32 operation. */
+    CRYPTO_ResultWidthSet(crypto, cryptoResult256Bits);
+
+    /* Clear sequence control registers */
+    crypto->SEQCTRL  = 0;
+    crypto->SEQCTRLB = 0;
+
+    /* Put the state back */
+    CORE_ENTER_CRITICAL();
+    CRYPTO_DDataWrite(&crypto->DDATA1, state);
+    CORE_EXIT_CRITICAL();
+
+    CRYPTO_EXECUTE_3( crypto,
+                      CRYPTO_CMD_INSTR_DDATA1TODDATA0,
+                      CRYPTO_CMD_INSTR_DDATA1TODDATA2,
+                      CRYPTO_CMD_INSTR_SELDDATA0DDATA1 );
+
+    /* Load the data block(s) */
+    for ( size_t i = 0; i < blocks; i++ ) {
+        if ((uint32_t)(&data[i*64]) & 0x3)
```

Comment:
Consider making a `#define` for `64`.

Reply:
Sure. Can I point out that the mbed TLS 'official' software implementation doesn't use a define for SHA blocksize, either?